addons/udes_stock_refactoring/tests/test_splitting.py

```diff
@@ -0,0 +1,647 @@
+        self.create_quant(
+            self.apple.id, self.test_stock_location_01.id, 10, package_id=apple_pallet.id
+        )
+
+        banana_pallet = self.create_package()
+        self.create_quant(
+            self.banana.id, self.test_stock_location_01.id, 10, package_id=banana_pallet.id
+        )
+
+        products_info = [{"product": self.apple, "qty": 10}, {"product": self.banana, "qty": 10}]
+        moves = self.create_move(self.picking, products_info)
+
+        apple_move = moves[0]
+        banana_move = moves[1]
+
+        self.picking.action_assign()
+
+        apple_pick = apple_move.picking_id
+        banana_pick = banana_move.picking_id
+
+        self.assertEqual(apple_pick.state, "assigned")
+        self.assertEqual(banana_pick.state, "assigned")
+        self.assertEqual(apple_pick.group_id.name, apple_pallet.name)
+        self.assertEqual(banana_pick.group_id.name, banana_pallet.name)
+
+        # Check we haven't mangled the moves or move_lines
+        apple_move = apple_pick.move_lines
+        self.assertEqual(len(apple_move), 1)
+        self.assertEqual(apple_move.product_uom_qty, 10)
+
+        apple_move_line = apple_pick.move_line_ids
+        self.assertEqual(len(apple_move_line), 1)
+        self.assertEqual(apple_move_line.product_qty, 10)
+        self.assertEqual(apple_move_line.package_id, apple_pallet)
+        self.assertEqual(apple_move_line.result_package_id, apple_pallet)
+        self.assertEqual(apple_move_line.product_id, self.apple)
+
+        banana_move = banana_pick.move_lines
+        self.assertEqual(len(banana_move), 1)
+        self.assertEqual(banana_move.product_uom_qty, 10)
+
+        banana_move_line = banana_pick.move_line_ids
+        self.assertEqual(len(banana_move_line), 1)
+        self.assertEqual(banana_move_line.product_qty, 10)
+        self.assertEqual(banana_move_line.package_id, banana_pallet)
+        self.assertEqual(banana_move_line.result_package_id, banana_pallet)
+        self.assertEqual(banana_move_line.product_id, self.banana)
+
+        # Check that the original picking has been reused
+        self.assertTrue(self.picking.exists())
+
+    def test_split_move(self):
+        """
+        Reserve self.picking with two pallet of the same product and check it
+        splits correctly when reserved.
+        """
+        MoveLine = self.env["stock.move.line"]
+
+        cherry_pallet1 = self.create_package()
+        self.create_quant(
+            self.cherry.id, self.test_stock_location_01.id, 10, package_id=cherry_pallet1.id
+        )
+        cherry_pallet2 = self.create_package()
+        self.create_quant(
+            self.cherry.id, self.test_stock_location_01.id, 10, package_id=cherry_pallet2.id
+        )
+
+        products_info = [{"product": self.cherry, "qty": 20}]
+        self.create_move(self.picking, products_info)
+
+        self.picking.action_assign()
+
+        cherry_pallet_move_lines = MoveLine.search(
```

Comment:
Can we not just get them from the picking?

Reply:
No because the original picking only has one move line, the other move line is in another picking that was created by refactoring.